js/packages/web/src/views/home/index.tsx

```diff
@@ -55,11 +56,14 @@ export const HomeView = () => {
   const liveAuctions = auctions
   .sort((a, b) => a.auction.info.endedAt?.sub(b.auction.info.endedAt || new BN(0)).toNumber() || 0);
 
+
   const items =
     activeKey === LiveAuctionViewState.All
       ? liveAuctions
       : activeKey === LiveAuctionViewState.Participated ?
       liveAuctions.concat(auctionsEnded).filter((m, idx) => m.myBidderMetadata?.info.bidderPubkey.toBase58() == wallet?.publicKey?.toBase58()):
+      activeKey === LiveAuctionViewState.Resale ? 
+      liveAuctions.filter((m, idx) => m.thumbnail.metadata.info.primarySaleHappened == true) :
```

Comment:
Given the number of states there are let's turn this into a switch statement. Also it's not just thumbnail. You want to actually check the items AuctionViewItem[][] struct. Iterate through each array, for each item in it, and check if any have this. It could be someone gets around this check by selling one original item and one non-original as secondary.

Reply:
You're also going to need to edit the heroAuction variable setup on line 45 to keep the hero from being set to one of these resales.

Reply:
Not clear on why would we need to loop through AuctionViewItem[][]. We are already loop through all live auctions and checking if that live auction is primary or not

Reply:
auctions can have more than one item on them... when you check thumbnail here, you are only checking one of the items. You know what, this is good enough honestly for 99% of auctions, and does save some extra cpu loops.

Reply:
Well in that case if any of the items in that auction is a resale, what would thumbnail presale status be set to. Also the auction would be displaying all items together, so if any item is a resale then the auction becomes partly resale and we can't display items separately, i think. Please let me know if my understanding is wrong.